### PR TITLE
TRTLLM new API support

### PR DESCRIPTION
**Pull request by meatybobby**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

There's no need to comment `jenkins` on the PR to trigger Jenkins CI.
The GitHub Actions CI will run automatically when the PR is opened.
To run CI on an untrusted fork, a NeMo user with write access must click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/export/trt_llm/tensorrt_llm_run.py [hunk 1]
```diff
@@ -55,7 +56,7 @@ class TensorrtLLMHostContext:
 class TensorrtLLMWorkerContext:
     """The MPI worker side context for TRT LLM inference."""
 
-    decoder: tensorrt_llm.runtime.GenerationSession = None
+    decoder: ModelRunner = None
```
Comment (jiemingz) [inline nemo/export/trt_llm/tensorrt_llm_run.py:59]: Maybe you could also add support for ModelRunnerCpp? It should be a very similar API, and the TRTLLM team tells us that the python backend is really only for testing, and for production the cpp backend should be used

File: nemo/export/trt_llm/tensorrt_llm_run.py [hunk 2]
```diff
@@ -210,60 +229,84 @@ def _forward(
...
+            #     bad_words_list=bad_words_list,
+            #     no_repeat_ngram_size=no_repeat_ngram_size,
+            #     streaming=streaming,
+            #     output_sequence_lengths=True,
+            #     return_dict=True,
+            # )
+
+            prompt_tasks = None if task_ids is None else ",".join(str(task) for task in task_ids)
+
+            outputs = decoder.generate(
```
Comment (cjluo-omniml) [inline nemo/export/trt_llm/tensorrt_llm_run.py:286]: I think you can just use the HLAPI like LLM to run inference. The API is much cleaner to call than this ModelRunnerCpp

**Comment by oyilmaz-nvidia** [comment]
Tested most of the models and getting this gibberish output from only the nemotron 22B base one 

```
--- Output:  [['is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is in the best way to get the best price for your car.\n\nHow to sell your car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car car'], ['is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is the best way to get to the hotel from the airport?\n\nThe hotel is located 1 km from the airport, which means that there is is is is is is is is is is is is is is is is is is is is is is is is is is'], ['is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is is the best way to get to the hotel from the airport?\n\nThe hotel is located 1 km from the airport.\n\nWhat is the best way way way way way way way way way way way way way way way way way way way way way way way way way']]
```

Can you guys please take a look at this?

**Review by oyilmaz-nvidia — APPROVED** [review]
Let's merge this PR and then we'll likely do a cleanup, basically we'll need to remove the parts where we don't use anymore.

**Comment by oyilmaz-nvidia** [comment]
jenkins

**Review by oyilmaz-nvidia — APPROVED** [review]

**Pull request merged**